### PR TITLE
cmd/testscript: add support to pass through env vars with -e

### PR DESCRIPTION
**Pull request by myitcv**

### CODE REVIEW

File: cmd/testscript/help.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ The testscript command runs github.com/rogpeppe/go-internal/testscript scripts
 in a fresh temporary work directory tree.
 
 Usage:
-    testscript [-v] files...
+    testscript [-v] [-e VAR] files...
```
Comment (rogpeppe) [inline cmd/testscript/help.go]: maybe:

    testscript [-v] [-e VAR]... files..

?

File: cmd/testscript/help.go [hunk 2]
```diff
@@ -31,6 +31,11 @@ proxy server. See the documentation for
 github.com/rogpeppe/go-internal/goproxytest for details on the format of these
 files/directories.
 
+Environment variables can be passed through to each script with the -e flag,
+where VAR is the name of the variable. Variables override testscript defined
```
Comment (rogpeppe) [inline cmd/testscript/help.go]: s/testscript defined/testscript-defined/

?

File: cmd/testscript/help.go [hunk 3]
```diff
@@ -31,6 +31,11 @@ proxy server. See the documentation for
...
+values, with the exception of WORK which cannot be be overridden. The -e flag
```
Comment (rogpeppe) [inline cmd/testscript/help.go]: s/be be/be/

File: cmd/testscript/help.go [hunk 4]
```diff
@@ -31,6 +31,11 @@ proxy server. See the documentation for
...
+can appear multiple times.
```
Comment (rogpeppe) [inline cmd/testscript/help.go]: s/can appear multiple times/can appear multiple times to specify multiple variables./

?

File: cmd/testscript/testdata/env_var.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Test passing environment variables to scripts
```
Comment (rogpeppe) [inline cmd/testscript/testdata/env_var.txt]: Just to sanity check, how about adding a .gomodproxy entry to file.txt and check that the env var entries from that are correctly overridden (e.g. GOARCH, GOPATH)?

File: cmd/testscript/help.go (hunk 4 above)
Comment (myitcv) [reply cmd/testscript/help.go]: Thanks.

File: cmd/testscript/help.go (hunk 3 above)
Comment (myitcv) [reply cmd/testscript/help.go]: Thanks.

File: cmd/testscript/help.go (hunk 2 above)
Comment (myitcv) [reply cmd/testscript/help.go]: Thanks

File: cmd/testscript/help.go (hunk 1 above)
Comment (myitcv) [reply cmd/testscript/help.go]: Thanks

File: cmd/testscript/testdata/env_var.txt (hunk 5 above)
Comment (myitcv) [reply cmd/testscript/testdata/env_var.txt]: Per our chat, completely agree. 

I've actually added a more comprehensive set of tests around this. Two files, one with `go` in the `PATH`, the other without. 

The matrix of scenarios tested is then:

* with and without `go` in `PATH
* with `.gomodproxy` and without
* with `GOPROXY` set in env and without
* with `-e` set and without

This also helps to confirm that setting a custom `GOPROXY` atop a `.gomodproxy` being present in the script results in the former, i.e. the override behaviour as described. 

The code before was just wrong; the vars were applied first when (obviously) they should have been applied last.

**Pull request merged**